client/src/main/App.tsx

```diff
@@ -1,12 +1,240 @@
 import React from 'react';
 import './App.css';
+import MuiAlert from '@material-ui/lab/Alert';
+import { Snackbar } from '@material-ui/core';
+import TopMenuBar from './components/TopMenuBar';
+import {
+  User,
+  AllUserData,
+  ProjectObjects,
+  TaskObjects,
+  UserSettings,
+} from './dbTypes';
 
-function App(): JSX.Element {
-  return (
-    <div className="App">
-      <header className="App-header">PointSpire</header>
-    </div>
-  );
+/**
+ * Used to determine the severity of an alert for the snackbar of the app.
+ */
+type SnackBarSeverity = 'error' | 'warning' | 'info' | 'success';
+
+type AppState = {
+  /**
+   * Determines if the toast window at the bottom is open.
+   */
+  snackBarOpen: boolean;
+
+  /**
+   * The current severity that was set for the toast window at the bottom.
+   */
+  snackBarSeverity: SnackBarSeverity;
+
+  /**
+   * The text for the toast window at the bottom.
+   */
+  snackBarText: string;
+
+  /**
+   * The primary user object in memory for the client.
+   */
+  user?: User;
+
+  /**
+   * All of the projects associated with the user.
+   */
+  projects?: ProjectObjects;
+
+  /**
+   * All of the tasks associated with the user.
+   */
+  tasks?: TaskObjects;
+};
+
+type AppProps = unknown;
+
+/**
+ * The base url for the server. This can be changed later to be the production
+ * url or be conditionally changed.
+ */
+const baseServerUrl = 'http://localhost:8055';
```

Comment:
mite want to change before merging to master

Reply:
Good call. Updated :)